src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -206,15 +206,19 @@ public SlaveTemplate(String ami, String zone, SpotConfiguration spotConfig, Stri
         this.tags = tags;
         this.idleTerminationMinutes = idleTerminationMinutes;
         this.associatePublicIp = associatePublicIp;
-        this.connectionStrategy = connectionStrategy;
+        if (connectionStrategy == null) {
```

Comment:
You could change all this line to just `this.connectionStrategy = connectionStrategy == null ? ConnectionStrategy.PRIVATE_IP : connectionStrategy;`